specs/core/0_beacon-chain.md

```diff
@@ -601,6 +601,21 @@ def is_slashable_validator(validator: Validator, epoch: Epoch) -> bool:
     return (not validator.slashed) and (validator.activation_epoch <= epoch < validator.withdrawable_epoch)
 ```
 
+#### `is_eligible_for_activation`
+
+```python
+def is_eligible_for_activation(state: BeaconState, validator: Validator) -> bool:
+    """
+    Check if ``validator`` is eligible for activation.
+    """
+    return (
+        # Was placed in activation queue prior to most recent finalized epoch
+        validator.activation_eligibility_epoch < state.finalized_checkpoint.epoch
```

Comment:
So instead of making a temporary activation that is overwritten when not within finality, we only consider eligible epochs directly? Doesn't this allow activations after the current finalized epoch to be scheduled, if finality takes longer than the activation delay?

Reply:
* We can move into the activation queue (set `activation_eligibility_epoch`) whenever the minimum balance threshold is hit
* We allow _removal_ from the queue and into activation, when the knowledge of being in the queue  is finalized
* When removed from activation queue, we set `activation_epoch` to be `compute_activation_exit_epoch(current_epoch)` (an epoch slightly in the future). This will happen regardless of subsequent finality. We are protected because the entrance into the queue was finalized and known

Reply:
1. Ah `<` makes much more sense...! 👍 
2. What's wrong with the `validator.activation_eligibility_epoch == state.finalized_checkpoint.epoch` case here? Isn't the `finalized_checkpoint.epoch` itself safe for activation?

Reply:
When we finalize epoch N, we are finalizing a block/state at the 0th block of N. The latest activation eligibilities happen at the end of an epoch and are set to current_epoch then. So the finalized activations at N are really those that were at the end of N-1.

I'm realizing, it might just make more sense to `validator.activation_eligibility_epoch = get_current_epoch(state) + 1` instead of doing this indirect `<`

Reply:
ahh you're right, it's from `compute_start_slot_at_epoch`.

`validator.activation_eligibility_epoch = get_current_epoch(state) + 1` would be more clear on what is happening here and fit for the naming. 👍